website/data/plugin-framework-nav-data.json

```diff
@@ -23,6 +23,10 @@
       }
     ]
   },
+  {
+    "title": "Protocol RPCs",
+    "path": "rpcs"
+  },
```

Comment:
Just to raise a slight concern: I'm a little worried that this page is a little _too_ detailed in its current form to be the first potential page of the navigation. The information is real nice (e.g. relating Terraform commands to how they interact with provider code), but I think it might completely overwhelm newer developers with a lot of implementation bits that are not fully necessary for them to successfully develop a provider. I'm not sure I have the best constructive feedback yet on how to address that potential concern though. 🙁 

One thing is that we currently have pages for each of the provider implementation details, such as Providers > Validating Configuration. Maybe some of this content could exist towards the bottom of those for advanced readers so its a little more self-contained depending on what a developer is wanting to do. Otherwise, I think we should consider how to link those pages back to this information in some form.

I don't think doing the above would fully alleviate the need for some of the higher level information of this page, but it might help quell the information overload issue by colocating the nitty-gritty RPC details as an "implementation details" section for each of the other pages that has a call out that this information is provided only for developers wanting a more technical deep dive.

Reply:
Might be best if we discuss how we want to proceed here. I'm more than happy to add information and/or links wherever you think would best serve the reader.

Reply:
Have refactored following discussion to include a new `Internals` section in the documentation and link to the `RPCs and Framework Functionality` page from other pages in the docs.